test/uk/gov/hmrc/merchandiseinbaggage/controllers/PreviousDeclarationDetailsControllerSpec.scala

```diff
@@ -0,0 +1,243 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.merchandiseinbaggage.controllers
+
+import play.api.test.Helpers._
+import uk.gov.hmrc.http.HttpClient
+import uk.gov.hmrc.merchandiseinbaggage.connectors.MibConnector
+import uk.gov.hmrc.merchandiseinbaggage.model.api.{Amendment, CategoryQuantityOfGoods, Country, Currency, DeclarationGoods, DeclarationId, DeclarationType, ExportGoods, GoodsVatRates, ImportGoods, JourneyDetails, JourneyInSmallVehicle, NotRequired, Paid, PaymentStatus, Port, PurchaseDetails, SessionId, YesNoDontKnow}
+import uk.gov.hmrc.merchandiseinbaggage.model.core.DeclarationJourney
+import uk.gov.hmrc.merchandiseinbaggage.stubs.MibBackendStub.givenPersistedDeclarationIsFound
+import uk.gov.hmrc.merchandiseinbaggage.views.html.{PreviousDeclarationDetailsView}
+import uk.gov.hmrc.merchandiseinbaggage.wiremock.WireMockSupport
+import uk.gov.hmrc.merchandiseinbaggage.config.MibConfiguration
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import java.time.{LocalDate, LocalDateTime}
+
+class PreviousDeclarationDetailsControllerSpec extends DeclarationJourneyControllerSpec with WireMockSupport with MibConfiguration {
+
+  val journeyDetails: JourneyDetails = JourneyInSmallVehicle(
+    Port("DVR", "title.dover", isGB = true, List("Port of Dover")),
+    LocalDate.now(),
+    "T5 RRY"
+  )
+
+  def createImport(desc: String, price: String): ImportGoods = ImportGoods(
+    CategoryQuantityOfGoods(desc, "1"),
+    GoodsVatRates.Twenty,
+    YesNoDontKnow.Yes,
+    PurchaseDetails(price, Currency("EUR", "title.euro_eur", Some("EUR"), List("Europe", "European")))
+  )
+
+  def createExport(desc: String, price: String): ExportGoods = ExportGoods(
+    CategoryQuantityOfGoods(desc, "1"),
+    Country("FR", "title.france", "FR", isEu = true, Nil),
+    PurchaseDetails(price, Currency("EUR", "title.euro_eur", Some("EUR"), List("Europe", "European")))
+  )
+
+  def createAmendment(desc: String, paymentStatus: Option[PaymentStatus]): Amendment = Amendment(
+    LocalDateTime.now,
+    DeclarationGoods(createImport(desc, "99.99") :: Nil),
+    None,
+    paymentStatus,
+    Some("Digital")
+  )
+
+  "check the view can" should {
+    "display 'continue' button when within date range" in {
+      val view = app.injector.instanceOf[PreviousDeclarationDetailsView]
+      val goods = createImport("wine", "99.99") :: createImport("cheese", "199.99") :: Nil
+      val result = view.render(goods, journeyDetails, DeclarationType.Import, withinTimerange = true, fakeRequest, messages, appConfig)
+      val html = contentAsString(result)
+
+      html must include(messageApi("previousDeclarationDetails.add_goods"))
+      html must include("govuk-button")
```

Comment:
we don't use `view.render` in our controller tests... also you can move these into our webdriver tests